### PR TITLE
Various fixes and enhancements

### PR DESCRIPTION
**Pull request by sofianehaddad**
- Fix Sample::sortUnique which provides default description even if intial sample provides a different one
- Perform VisualTest::DrawEmpiricalCDF (xMin/xMax are now optional)
- In CovarianceModel, fix some docstrings, setParameter method; Made also {g,s}etFullParameter and getFullParameterDescription public

### CODE REVIEW

**Comment by regislebrun** [comment]
+1

**Comment by regislebrun** [comment]
You mean 'pubic' or 'public' in commit a4318b4 ;-)?
+1

File: lib/src/Base/Stat/CovarianceModelImplementation.cxx [hunk 1]
```diff
@@ -574,14 +574,24 @@ void CovarianceModelImplementation::setFullParameter(const Point & parameter)
   // Here we manage only the generic parameters
   // First the scale parameter
   UnsignedInteger index = 0;
+  // Check the size
+  const UnsignedInteger totalSize = spatialDimension_ + dimension_ * (dimension_ + 1) / 2;
+  if (parameter.getSize() < totalSize)
+    throw InvalidArgumentException(HERE) << "In CovarianceModelImplementation::setFullParameter, points have incompatible size. Point size = " << parameter.getSize()
+                                         << " wheras expected size = " << totalSize ;
```
Comment (dbarbier) [inline lib/src/Base/Stat/CovarianceModelImplementation.cxx]: Typo: whereas

File: lib/src/Base/Stat/CovarianceModelImplementation.cxx [hunk 2]
```diff
@@ -574,14 +574,24 @@ void CovarianceModelImplementation::setFullParameter(const Point & parameter)
...
+
   for (UnsignedInteger i = 0; i < spatialDimension_; ++ i)
   {
+    if (parameter[index] <= 0.0)
```
Comment (dbarbier) [inline lib/src/Base/Stat/CovarianceModelImplementation.cxx]:
```
if (!(parameter[index] > 0.0))
```

**Comment by jschueller** [comment]
please mention #891 in the changelog

**Comment by sofianehaddad** [comment]
@jschueller @dbarbier thanks, done.

**Pull request merged**